### PR TITLE
Dev tool to migrate hints.yaml files to v1 specification

### PR DESCRIPTION
**Pull request by raoulmillais**
Adds a devtool to help with transforming the schema of hints files en-masse.

Notes:

* Written in JavaScript for the sake of getting it done quickly (maybe should be part of the simulator golang binary eventually
* Need to be careful of the fact that scenario validation now happens in `perturb.sh` as well as this separate tool (incidentally)
* Does not currently nest hints based on the task they apply to as the hints as they stand are not consistent and I need to deal with whitespace in eg.  "Task 2"
* Transformation logic is unit tested ~~but not tested on the real scenarios~~, I will do that as part of a separate PR to transform all the scenarios, and so if I discover problems I may amend this PR.
* Migrates (updated) specification of the `hints.yaml` format notes off cryptpad
* Documents usage

**Look at the unit test file for the transformation**

Screenshot from 2019-10-31 16-12-08
Before and after running the script on a `hints.yaml` file

To test locally:

* checkout this branch
* `make devtools`
* `migrate-hints`
* Take a look at the changes in the `simulation-scripts` folder

### CODE REVIEW

**Comment by raoulmillais** [comment]
Screenshot from 2019-10-31 16-12-08
Before and after running the script on a `hints.yaml` file

**Comment by raoulmillais** [comment]
I think we ought to also transform `general_overview` to `general-overview` for consistency

**Comment by raoulmillais** [comment]
also maybe we should go back to calling them `scenario.yaml` since they contain more than just hints (and soon will have scoring too)

File: Makefile [hunk 1]
```diff
@@ -30,6 +30,10 @@ all: test
 setup-dev: ## Initialise simulation tree with git hooks
 	@ln -s $(shell pwd)/setup/hooks/pre-commit $(shell pwd)/.git/hooks/pre-commit
 
+.PHONY: dep
```
Comment (abdullahgarcia) [inline Makefile]: Is this purposely called "dep" or is it a typo for "dev"?

Comment (abdullahgarcia) [reply Makefile]: Shouldn't it be "devtools"?

Comment (raoulmillais) [reply Makefile]: copy and paste error! :) thanks

File: tools/migrate-hints/test/transforms_test.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }, {
+      hints: {
+        "hint-1": "test hint 1",
+        "hint-2": "test hint 2"
+      }
+    }
+  ]
+
+  const expected = [{
+      kind: "cp.simulator/scenario:1.0.0",
```
Comment (abdullahgarcia) [inline tools/migrate-hints/test/transforms_test.js]: How is this parameter delivered?

I can see in the format the reference "semver-version", but I cannot pin it back to the test.

Comment (raoulmillais) [reply tools/migrate-hints/test/transforms_test.js]: it's just a hardcoded string in the transform function.  I've arbirtrarily chosen v1.0.0.  This is the version of the scema and not of simulator itself.  We can use it to detect old scenarios, automate upgrading etc in the future if we wish

**Pull request merged**